lib/fieldhand/datestamp.rb

```diff
@@ -12,16 +16,25 @@ def self.parse(datestamp)
       end
     end
 
+    # Return a string UTC datestamp given a string, `Date` or `Time`.
+    #
+    # The granularity of the resulting datestamp depends on the input type:
+    #
+    # * Strings are returned untouched (assuming they are already formatted datestamps)
+    # * Dates will return a date-level granularity datestamp, e.g. 2001-01-01
+    # * Times will return a time-level granularity UTC datestamp, e.g. 2001-01-01T00:00:00Z
+    # * DateTimes will return a time-level granularity UTC datestamp, e.g. 2001-01-01T00:00:00Z
+    # * Anything else is assumed to respond to `to_time`, `utc` or `xmlschema`
     def self.unparse(datestamp)
       case datestamp
       when ::String
         datestamp
-      when ::Date
-        datestamp.strftime
+      when ::DateTime
+        unparse(::Time.xmlschema(datestamp.to_s))
       when ::Time
         datestamp.utc.xmlschema
       else
-        datestamp.xmlschema
+        datestamp.strftime
```

Comment:
If we don't have a specific case for the type of `datestamp`, perhaps we should default to `xmlschema` rather than `strftime` as it conveys the most information.